scripts/staging/knn.dml

```diff
@@ -330,66 +334,45 @@ getKNeighbor = function (  matrix[double] in_m_data,
 
 getErr_k = function (  matrix[double] in_m_neighbor_value,
         matrix[double] in_m_cl,
-        integer in_i_cl_type , 
-        integer in_i_k_min  ) 
+        integer in_i_cl_type ,
+        integer in_i_k_min  )
     return (  matrix[double] out_m_err  ){
     i_col = ncol( in_m_neighbor_value  );
-    i_row  = nrow( in_m_neighbor_value  );  
-    
+    i_row  = nrow( in_m_neighbor_value  );
+
     out_m_err = matrix( 0,i_row,i_col - in_i_k_min + 1  );
     if( in_i_cl_type == 2  ){
         #category
-        m_correct = in_m_neighbor_value != in_m_cl[1:i_row,];
+        m_correct = ppred( in_m_neighbor_value,in_m_cl[1:i_row,],"!="  );
```

Comment:
`ppred` has been deprecated in the time since this PR was created.  This won't be merged until this and other usages of `ppred` are updated to the correct syntax (in red).